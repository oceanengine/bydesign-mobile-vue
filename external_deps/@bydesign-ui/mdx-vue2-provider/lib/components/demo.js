/**
 * Component Demo
 */
import createBEM from '../utils/bem';
import HL from './highlight';
import { openCodepen, openSandbox, copyToClipboard } from '../utils/editor';
var bem = createBEM('demo');

var Demo = function Demo() {
  return {
    props: {
      title: [String, Number],
      desc: [String, Number],
      code: String,
      src: String,
      srcRender: Function,
      maxHeight: Number
    },
    data: function data() {
      return {
        unfolded: false
      };
    },
    methods: {
      setUnfolded: function setUnfolded(value) {
        this.unfolded = value;
      }
    },
    render: function render() {
      var h = arguments[0];
      var title = this.title,
          desc = this.desc,
          code = this.code,
          srcRender = this.srcRender,
          unfolded = this.unfolded,
          setUnfolded = this.setUnfolded,
          _this$maxHeight = this.maxHeight,
          maxHeight = _this$maxHeight === void 0 ? 200 : _this$maxHeight;
      var containerStyle = {
        maxHeight: "".concat(maxHeight, "px")
      };
      return h("div", {
        "class": bem()
      }, [h("div", {
        "class": bem('title'),
        "attrs": {
          "id": title.toString()
        }
      }, [title]), srcRender !== undefined && h("div", {
        "class": bem('box')
      }, [h("div", {
        "class": bem('show')
      }, [srcRender()]), desc && h("div", {
        "class": bem('desc')
      }, [desc]), code !== undefined && h("div", {
        "class": bem('code-tool')
      }, [h("div", {
        "class": bem('operations')
      }, [h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openCodepen(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/codepen.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            openSandbox(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/sandbox.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            copyToClipboard(unescape(code));
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/copy.svg')
        }
      }), h("span", {
        "class": bem('operation-btn'),
        "on": {
          "click": function click() {
            setUnfolded(!unfolded);
          }
        },
        "domProps": {
          "innerHTML": require('!html-loader!../components/icons/code.svg')
        }
      })])]), code !== undefined && unfolded && h("div", {
        "class": bem('code'),
        "style": containerStyle
      }, [h(HL, {
        "attrs": {
          "code": unescape(code)
        }
      })])])]);
    }
  };
};

export default Demo;