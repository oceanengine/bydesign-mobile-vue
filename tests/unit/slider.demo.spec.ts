import Demo from '@src/components/slider/demo/index.vue';
import { snapshotDemo } from '../demo';

snapshotDemo(Demo);
