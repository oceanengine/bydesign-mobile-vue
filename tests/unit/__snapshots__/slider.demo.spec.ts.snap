// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders demo correctly 1`] = `
<div class="demo">
  <div class="byted-header--wrapper">
    <div class="byted-header byted-header--default byted-header--fixed byted-hairline--bottom">
      <div class="byted-header__left"><i class="byted-header__icon" style="width: 1.2rem; height: 1.2rem;">test-file-stub</i></div>
      <div class="byted-header__title"><span class="demo-header-icon"><img src="https://s2.pstatp.com/cdn/expire-1-d/byted-creative-app/static/ByDesign-logo.svg" class="img-icon"></span></div>
    </div>
  </div>
  <div><span class="byted-icon byted-icon-module-slider" style="font-size: 20px;"><svg width="20" height="20" viewBox="0 0 64 64" fill="none" xmlns="http://www.w3.org/2000/svg"><path d="M16 7H39.1429H48V23H39.1429H36L32 27L28 23H16V7Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M36 39L58 39" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M6 39L28 39" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M25 53L40 53" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M48 53H58" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M6 53H17" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M48 53C48 55.2091 46.2091 57 44 57C41.7909 57 40 55.2091 40 53C40 50.7909 41.7909 49 44 49C46.2091 49 48 50.7909 48 53Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M26 15H38" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><path d="M25 53C25 55.2091 23.2091 57 21 57C18.7909 57 17 55.2091 17 53C17 50.7909 18.7909 49 21 49C23.2091 49 25 50.7909 25 53Z" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></path><circle cx="32" cy="39" r="4" fill="transparent" stroke="#333" stroke-width="4" stroke-linecap="round" stroke-linejoin="round"></circle></svg></span></div>
  <div>
    <div class="byted-slider">
      <div class="byted-slider__bar" style="width: 50%;">
        <div class="byted-slider__bar__wrapper">
          <div class="byted-slider__button"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="byted-slider">
      <div class="byted-slider__bar" style="width: 50%;">
        <div class="byted-slider__bar__wrapper">
          <div class="byted-slider__button"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="byted-slider">
      <div class="byted-slider__bar" style="width: 50%;">
        <div class="byted-slider__bar__wrapper">
          <div class="byted-slider__button"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="byted-slider">
      <div class="byted-slider__bar" style="width: 40%;">
        <div class="byted-slider__bar__wrapper">
          <div class="byted-slider__button"></div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div class="byted-slider byted-slider--disabled">
      <div class="byted-slider__bar" style="width: 50%;">
        <div class="byted-slider__bar__wrapper">
          <div class="byted-slider__button"></div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
